File: core/sdk/dotnet/Models/Entities/Abstract/BaseEntity.cs
using Microsoft.Extensions.Logging;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Agience.Core.Models.Entities.Abstract
{
    public abstract class BaseEntity
    {
        [JsonPropertyName("id")]
        public string? Id { get; set; }

        [JsonPropertyName("created_date")]
        public DateTime? CreatedDate { get; set; }

        [JsonIgnore]
        [NotMapped]
        public Dictionary<string, object?> Metadata { get; set; } = new();

    }
}

File: core/sdk/python/models/entities/abstract/base_entity.py
from datetime import datetime
from typing import Dict, Optional
from pydantic import BaseModel, Field


class BaseEntity(BaseModel):
    id: Optional[str] = None
    created_date: Optional[datetime] = None
    metadata: Dict[str, Optional[object]] = Field(
        default_factory=dict, exclude=True)
